### PR TITLE
Filter books by genre

### PR DESCRIPTION
**Pull request by andyw8**
https://trello.com/b/rxNcTk1k/ruby-bookshelf

### CODE REVIEW

**Comment by pedrosmmoreira** [comment]
:+1: Good work! Couple of suggestions: how do you feel about pulling out a `FilterController`? Also, since  we already have a `Filter` model, how do you feel about pulling the reference to `Genre` from the form view into a method on that class?

**Comment by andyw8** [comment]
@pedrosmmoreira Updated as per your suggestion. I was bit unsure about the resource nesting, let me know what you think.

File: config/routes.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 Rails.application.routes.draw do
   root "books#index"
 
-  resources :books, only: [:index, :show]
+  resources :books, only: [:index, :show] do
+    collection do
+      resource :filter, only: :show, as: :book_filter, controller: "books/filter"
```
Comment (andyw8) [inline config/routes.rb]: I'd prefer to keep this on one line to keep the routes uniform.

**Comment by taylorrf** [comment]
It's okay :+1:

**Comment by andyw8** [comment]
Sorry, I forgot we agreed to not use the GitHub merge button.

**Pull request merged**